src/sentry/static/sentry/app/views/projectDetail/projectScoreCards/projectStabilityScoreCard.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+
+import AsyncComponent from 'app/components/asyncComponent';
+import {canIncludePreviousPeriod, getDiffInMinutes} from 'app/components/charts/utils';
+import {getParams} from 'app/components/organizations/globalSelectionHeader/getParams';
+import ScoreCard from 'app/components/scoreCard';
+import {DEFAULT_STATS_PERIOD} from 'app/constants';
+import {t} from 'app/locale';
+import {GlobalSelection, Organization} from 'app/types';
+import {defined, percent} from 'app/utils';
+import {formatAbbreviatedNumber} from 'app/utils/formatters';
+import {getPeriod} from 'app/utils/getPeriod';
+import {displayCrashFreePercent, getCrashFreePercent} from 'app/views/releases/utils';
+
+import MissingReleasesButtons from '../missingFeatureButtons/missingReleasesButtons';
+
+type SessionApiResponse = {
+  query: string;
+  intervals: string[];
+  groups: {
+    by: Record<string, string>;
+    totals: Record<string, number>;
+    series: Record<string, number[]>;
+  }[];
+};
+
+type Props = AsyncComponent['props'] & {
+  organization: Organization;
+  selection: GlobalSelection;
+};
+
+type State = AsyncComponent['state'] & {
+  currentSessions: SessionApiResponse | null;
+  previousSessions: SessionApiResponse | null;
+  noSessionEver: boolean;
+};
+
+class ProjectStabilityScoreCard extends AsyncComponent<Props, State> {
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      currentSessions: null,
+      previousSessions: null,
+      noSessionEver: false,
+    };
+  }
+
+  getEndpoints() {
+    const {organization, selection} = this.props;
+
+    const {projects, environments, datetime} = selection;
+    const {period, start, end} = datetime;
+    const commonQuery = {
+      environment: environments,
+      project: projects[0],
+      field: 'sum(session)',
+      groupBy: 'session.status',
+      interval: getDiffInMinutes(datetime) >= 24 * 60 ? '1d' : '1h',
+    };
+
+    // Unfortunately we can't do something like statsPeriod=28d&interval=14d to get scores for this and previous interval with the single request
+    // https://github.com/getsentry/sentry/pull/22770#issuecomment-758595553
+
+    const endpoints: ReturnType<AsyncComponent['getEndpoints']> = [
+      [
+        'currentSessions',
+        `/organizations/${organization.slug}/sessions/`,
+        {
+          query: {
+            ...commonQuery,
+            ...getParams(datetime),
+          },
+        },
+      ],
+    ];
+
+    if (!start && !end && canIncludePreviousPeriod(true, period)) {
```

Comment:
you often check if start and end are defined, should we create a simple function to check this?